doc/usage/restructuredtext/domains.rst

```diff
@@ -113,6 +113,29 @@ In short:
   refer to ``Queue.Queue.get`` but only display ``get`` as the link text.
 
 
+Table of Contents
+~~~~~~~~~~~~~~~~~
+
+The directive option flag ``:section:`` wraps a description in a
+section. Like other sections, the object will have an entry in the
+table of contents. For example::
+  .. cpp::function int fun1(int a)
+     :section:
+
+  .. cpp::function int fun2(int a)
+     :section:
+
+Is the same as::
+
+  fun1
+  -----
+  .. cpp::function int fun1(int a)
```

Comment:
My goal is to put the function in the TOC. I will look at doing it directly instead of creating the section. 

I am working on a big doc with multiple parts so having a single global config would not work. There needs to be a way to turn on the option for a region of the doc, but not all.

Closing the pull request while I investigate some more